### PR TITLE
feat: New design for adding people

### PR DESCRIPTION
**Pull request by przemvs**
https://wearezeta.atlassian.net/browse/ACC-200

Added new design for selected people in adding to the group.

image

### CODE REVIEW

File: src/script/components/UserList/UserList.tsx [hunk 1]
```diff
@@ -218,36 +229,86 @@ const UserList: React.FC<UserListProps> = ({
...
-              showArrow={showArrow}
-            />
-          </li>
-        ))}
-      </ul>
+      <>
+        {hasSelectedUsers && (
+          <>
+            <button onClick={() => toggleFolder(UserListFolder.SELECTED_CONTACTS)} css={collapseButton}>
+              <span css={collapseIcon(isSelectedContactsOpen)}>
```
Comment (arjita-mitra) [inline src/script/components/UserList/UserList.tsx]: Don't we need aria-hidden here?

File: src/script/components/UserList/UserList.tsx [hunk 2]
```diff
@@ -218,36 +229,86 @@ const UserList: React.FC<UserListProps> = ({
...
+                  const isLastItem = index === selectedUsers.length - 1;
+
+                  return renderParticipantItem(user, isLastItem);
+                })}
+            </ul>
+          </>
+        )}
+
+        <button onClick={() => toggleFolder(UserListFolder.CONTACTS)} css={collapseButton}>
+          <span css={collapseIcon(isContactsOpen)}>
```
Comment (arjita-mitra) [inline src/script/components/UserList/UserList.tsx]: Don't we need aria-hidden here?

File: src/script/components/UserList/UserList.test.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+let conversationRepository: ConversationRepository;
+
+beforeAll(() => {
+  testFactory.exposeConversationActors().then(factory => {
+    conversationRepository = factory;
+    return conversationRepository;
+  });
+});
+
+describe('UserList', () => {
```
Comment (arjita-mitra) [inline src/script/components/UserList/UserList.test.tsx:42]: Can you add some test for selected users functionality?

**Review by arjita-mitra — CHANGES_REQUESTED** [review]

File: src/script/components/UserSearchableList.tsx [hunk 4]
```diff
@@ -27,7 +27,7 @@ import {useKoSubscribableChildren} from 'Util/ComponentUtil';
 import {t} from 'Util/LocalizerUtil';
 import {sortByPriority} from 'Util/StringUtil';
 
-import {UserList} from './UserList';
+import {UserList} from './UserList/UserList';
```
Comment (PatrykBuniX) [inline src/script/components/UserSearchableList.tsx]: Can't we shorten it to just `./UserList` as it was before?

Comment (przemvs) [reply src/script/components/UserSearchableList.tsx]: Hmm.. smth wrong with linter, i had error with this import.. but fixed :)

File: src/script/components/UserList/UserList.test.tsx (hunk 3 above)
Comment (przemvs) [reply src/script/components/UserList/UserList.test.tsx:42]: added

File: src/script/components/list/ParticipantItem.tsx [hunk 5]
```diff
@@ -28,7 +28,7 @@ import {AvailabilityState} from 'Components/AvailabilityState';
 import {Avatar, AVATAR_SIZE} from 'Components/Avatar';
 import {ParticipantMicOnIcon} from 'Components/calling/ParticipantMicOnIcon';
 import {Icon} from 'Components/Icon';
-import {UserlistMode} from 'Components/UserList';
+import {UserlistMode} from 'Components/UserList/UserList';
```
Comment (atomrc) [inline src/script/components/list/ParticipantItem.tsx]:
```suggestion
import {UserlistMode} from 'Components/UserList';
```

File: src/script/page/LeftSidebar/panels/StartUI/PeopleTab.tsx [hunk 6]
```diff
@@ -27,7 +27,7 @@ import {partition} from 'underscore';
 import {WebAppEvents} from '@wireapp/webapp-events';
 
 import {Icon} from 'Components/Icon';
-import {UserList, UserlistMode} from 'Components/UserList';
+import {UserList, UserlistMode} from 'Components/UserList/UserList';
```
Comment (atomrc) [inline src/script/page/LeftSidebar/panels/StartUI/PeopleTab.tsx]:
```suggestion
import {UserList, UserlistMode} from 'Components/UserList';
```

Isn't there a `index.js` file in there?

File: src/script/components/UserList/UserList.tsx [hunk 7]
```diff
@@ -17,31 +17,37 @@
...
+import {UserState} from '../../user/UserState';
+import {InViewport} from '../utils/InViewport';
 
 export enum UserlistMode {
   COMPACT = 'UserlistMode.COMPACT',
   DEFAULT = 'UserlistMode.DEFAULT',
   OTHERS = 'UserlistMode.OTHERS',
 }
 
+export enum UserListFolder {
```
Comment (atomrc) [inline src/script/components/UserList/UserList.tsx]: Let's call those `sections` rather than `folders`

**Review by V-Gira — APPROVED** [review]

**Comment by przemvs** [comment]
@atomrc I talked with Jann, and he said to merge this task to edge branch. Message from Jann:

> Could this be merged to edge or another env first? This way we would have a much better overview which tests are affected and will break

**Comment by atomrc** [comment]
@przemvs sure, feel free to deploy to `edge` using this action https://github.com/wireapp/wire-webapp/actions/workflows/deploy-to-test-env.yml

**Comment by corodiak** [comment]
Looks good from QA 👍

**Review by arjita-mitra — APPROVED** [review]

**Pull request merged**